name_node/controllers/outer/upload_request_controller.go

```diff
@@ -0,0 +1,66 @@
+package outer
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+
+	namenode "github.com/SayedAlesawy/Videra-Storage/name_node"
+	"github.com/SayedAlesawy/Videra-Storage/utils/errors"
+	"github.com/julienschmidt/httprouter"
+)
+
+// ServeUploadURL Handles client's request to retrieve data node url for uploading
+func (server *Server) ServeUploadURL(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	// TODO: Add check if current name node is not the master and return the master ip
```

Comment:
I don't understand this. I mean if the name node is receiving requests, then it should serve it. It would be the responsibility of the client to try the standby name node if the master name node is failing requests, probably 503.

Reply:
We will have to think more about this, but let's assume that if a request went through, then it will be served by this instance.